### PR TITLE
Change parse/1 to return {decimal, binary} | :error

### PR DESCRIPTION
**Pull request by wojtekmach**

### CODE REVIEW

File: test/decimal_test.exs [hunk 1]
```diff
@@ -9,6 +9,70 @@ defmodule DecimalTest do
...
+    assert Decimal.parse("0.0") == {d(1, 0, -1), ""}
+    assert Decimal.parse("-0.0") == {d(-1, 0, -1), ""}
+    assert Decimal.parse("+0.0") == {d(1, 0, -1), ""}
+
+    assert Decimal.parse("0.0.0") == {d(1, 0, -1), ".0"}
+    assert Decimal.parse("-0.0.0") == {d(-1, 0, -1), ".0"}
+    assert Decimal.parse("+0.0.0") == {d(1, 0, -1), ".0"}
+
+    assert Decimal.parse("0.") == :error
+    assert Decimal.parse(".0") == :error
```
Comment (wojtekmach) [inline test/decimal_test.exs]: both of these values used to be allowed, not sure if we should keep them valid.

Comment (ericmj) [reply test/decimal_test.exs]: They should be allowed according to the spec: http://speleotrove.com/decimal/daconvs.html. What made them break?

Comment (wojtekmach) [reply test/decimal_test.exs]: Okay, I'll bring this behaviour back. The breakage was due to trying to simplifying parsing code.

Comment (wojtekmach) [reply test/decimal_test.exs]: updated!

**Pull request merged**